subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/configuration/InstrumentationParameters.kt

```diff
@@ -7,12 +7,18 @@ data class InstrumentationParameters(
     val initialParameters: Map<String, String> = emptyMap()
 ) : Map<String, String> by initialParameters, Serializable {
 
-    fun applyParameters(newParameters: Map<String, String>): InstrumentationParameters =
-        InstrumentationParameters(
+    fun applyParameters(newParameters: Map<String, String>): InstrumentationParameters {
+        newParameters.forEach { key, value ->
```

Comment:
it's not clear to me that instrumentation args are immutable from this point,
i'd expect check as close as possible to runner, but before any reservation job (to prevent resource waste)
but fail fast is also good idea. 
Maybe both? two checks?

Reply:
I don't want to know about it in many places.
My view is we have a structure InstrumentationParameters and it will guarantee that parameters isn't blank. If we had problems in future with incorrect reusing we will fix this.
Now is constructor parameters validation is missing i will add it

Reply:
we modify params once more at least here : https://github.com/avito-tech/avito-android/blob/12b0fdff3962329c8814112a726abd24efc15d3d/subprojects/gradle/runner/service/src/main/kotlin/com/avito/runner/service/worker/device/adb/AdbDevice.kt#L80

But i got your point, it's ok enough for now